### PR TITLE
feat(vertico): add tramp sources to consult-dir

### LINKED ISSUE #6258: Add consult-tramp
### Describe your request

I think [consult-tramp](https://github.com/Ladicle/consult-tramp) might be a welcome addition to Doom. It lists Tramp targets such as the hosts in the ssh configuration file and Docker containers.

### Briefly explain its use-case

1. Press `SPC f t`.
2. `consult-tramp` is run.
3. The minibuffer lists different Tramp targets, which the user can filter through and open.

### PR DESCRIPTION
**Pull request by elken**
Closes #6258

Could maybe do with some nicer Marginalia annotations, but it functions as expected.

### CODE REVIEW

File: modules/completion/vertico/packages.el [hunk 1]
```diff
@@ -12,6 +12,9 @@
 (package! consult-dir :pin "08f543ae6acbfc1ffe579ba1d00a5414012d5c0b")
 (when (featurep! :checkers syntax)
   (package! consult-flycheck :pin "0ad7e8ff15683a4d64b79c29b3fcf847edfe244b"))
+(package! consult-tramp
+        :recipe (:host github :repo "Ladicle/consult-tramp")
+        :pin "befa62baca768caa457b167e773b91f1bc7d661f")
```
Comment (hlissner) [inline modules/completion/vertico/packages.el:17]: Correct indentation (should be 2 spaces).

File: modules/completion/vertico/config.el [hunk 2]
```diff
@@ -158,6 +158,14 @@ orderless."
          ("C-x C-d" . consult-dir)
          ("C-x C-j" . consult-dir-jump-file)))
 
+(use-package! consult-tramp
+  :after consult
+  :init
+  (setq consult-tramp-method tramp-default-method)
+  (setq consult-tramp-enable-docker (featurep! :tools docker))
+  (map! :leader
+        :desc "Select TRAMP target" "f t" #'consult-tramp))
```
Comment (hlissner) [inline modules/completion/vertico/config.el:167]: Move to `config/default/+evil-bindings.el` and mirror to `config/default/+emacs-bindings.el`

File: modules/completion/vertico/config.el [hunk 3]
```diff
@@ -158,6 +158,14 @@ orderless."
          ("C-x C-d" . consult-dir)
          ("C-x C-j" . consult-dir-jump-file)))
 
+(use-package! consult-tramp
+  :after consult
+  :init
```
Comment (hlissner) [inline modules/completion/vertico/config.el:163]: `s/:init/:config`.

File: modules/completion/vertico/config.el [hunk 4]
```diff
@@ -158,6 +158,14 @@ orderless."
          ("C-x C-d" . consult-dir)
          ("C-x C-j" . consult-dir-jump-file)))
 
+(use-package! consult-tramp
+  :after consult
```
Comment (hlissner) [inline modules/completion/vertico/config.el:162]: Replace `:after consult` with `:defer t`. This package's sole entry point [is autoloaded](https://github.com/Ladicle/consult-tramp/blob/main/consult-tramp.el#L114), so it doesn't need to be loaded early.

**Review by hlissner — CHANGES_REQUESTED** [review]

**Comment by minad** [comment]
Hey @elken, I think consult-tramp should actually be completion-tramp (see https://github.com/Ladicle/consult-tramp/issues/1), since in principle this works with arbitrary completion packages. Or are there plans to take advantage of actual Consult functionality (async, preview, narrowing)?

**Comment by elken** [comment]
> Hey @elken, I think consult-tramp should actually be completion-tramp (see [Ladicle/consult-tramp#1](https://github.com/Ladicle/consult-tramp/issues/1)), since in principle this works with arbitrary completion packages. Or are there plans to take advantage of actual Consult functionality (async, preview, narrowing)?

I see your point and I actually do agree, the package isn't mine but it definitely seems like there's scope for improvement.

I'll amend Henrik's review (it was quite a rushed PR, in hindsight) and mull over other consult features to add in.

**Comment by elken** [comment]
I think once [this](https://github.com/Ladicle/consult-tramp/pull/2) has been merged this should also be fine to merge (pending another Henrik review)

**Comment by elken** [comment]
Converted to draft for now until the future of these changes is decided

**Comment by elken** [comment]
Once https://github.com/karthink/consult-dir/pull/16 is merged, `consult-dir` will need a bump; making this PR redundant.

Would it be worth amending this to have `SPC f t` open a pre-filtered version of `consult-dir` or just closing this?

**Review by iyefrat — CHANGES_REQUESTED** [review]
Looks good, just a few requests:

1. the bump commit summary should be `bump: consult-dir`, since it just bumps that package.
2. The readme changes should be in a separate PR to the rewrite docs branch

**Comment by elken** [comment]
> Looks good, just a few requests:
> 
>     1. the bump commit summary should be `bump: consult-dir`, since it just bumps that package.

That's what `doom/commit-bumps` filled out, but sure no worries

>     2. The readme changes should be in a separate PR to the rewrite docs branch

Yeah you're right, they're not strictly needed for this so I'll move them out

**Review by iyefrat — APPROVED** [review]

File: modules/completion/vertico/config.el [hunk 5]
```diff
@@ -183,8 +183,33 @@ orderless."
...
+        (let ((hosts)
+              (docker-tramp-use-names t))
+          (dolist (cand (docker-tramp--parse-running-containers))
+            (let ((user (unless (string-empty-p (car cand))
+                          (concat (car cand) "@")))
+                  (host (car (cdr cand))))
+              (push (concat "/docker:" user host ":/") hosts)))
+          hosts)))
+
+    (defvar consult-dir--source-tramp-docker
```
Comment (iyefrat) [inline modules/completion/vertico/config.el:201]: These don't follow the doom naming conventions

**Review by iyefrat — CHANGES_REQUESTED** [review]

File: modules/completion/vertico/config.el (hunk 5 above)
Comment (elken) [reply modules/completion/vertico/config.el:201]: You're right, I verbatim copied them from my README changes

Comment (elken) [reply modules/completion/vertico/config.el]: How's that?

Comment (iyefrat) [reply modules/completion/vertico/config.el]: :+1:

File: modules/completion/vertico/config.el [hunk 6]
```diff
@@ -227,11 +253,11 @@ orderless."
...
-    "Keymap for Embark package actions for packages installed by Doom."
-    ("h" doom/help-packages)
-    ("b" doom/bump-package)
-    ("c" doom/help-package-config)
-    ("u" doom/help-package-homepage))
+                        "Keymap for Embark package actions for packages installed by Doom."
+                        ("h" doom/help-packages)
+                        ("b" doom/bump-package)
+                        ("c" doom/help-package-config)
+                        ("u" doom/help-package-homepage))
```
Comment (iyefrat) [inline modules/completion/vertico/config.el:260]: Just fix this bit (you need to load embark before you format)

Comment (elken) [reply modules/completion/vertico/config.el]: Ah yeah, skimmed that my bad

**Review by iyefrat — APPROVED** [review]

**Review by hlissner — APPROVED** [review]

**Pull request merged**